dataproc/src/main/java/SubmitHadoopFSJob.java

```diff
@@ -0,0 +1,93 @@
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class SubmitHadoopFSJob {
+
+    public static ArrayList<String> stringToList(String s) {
+        return new ArrayList<>(Arrays.asList(s.split(" ")));
+    }
+
+    public static void submitHadoopFSQuery() throws IOException, InterruptedException {
+        // TODO(developer): Replace these variables before running the sample.
+        String projectId = "your-project-id";
+        String region = "your-project-region";
+        String clusterName = "your-cluster-name";
+        String hadoopFSQuery = "your-hadoop-fs-query";
+        submitHadoopFSJob(projectId, region, clusterName, hadoopFSQuery);
+    }
+
+    public static void submitHadoopFSJob(
+            String projectId, String region, String clusterName, String hadoopFSQuery)
+            throws IOException, InterruptedException {
+        String myEndpoint = String.format("%s-dataproc.googleapis.com:443", region);
+
+        // Configure the settings for the job controller client.
+        JobControllerSettings jobControllerSettings =
+                JobControllerSettings.newBuilder().setEndpoint(myEndpoint).build();
+
+        // Create a job controller client with the configured settings. Using a try-with-resources closes the client,
+        // but this can also be done manually with the .close() method.
+        try (JobControllerClient jobControllerClient =
+                     JobControllerClient.create(jobControllerSettings)) {
+
+            // Configure cluster placement for the job.
+            JobPlacement jobPlacement = JobPlacement.newBuilder().setClusterName(clusterName).build();
+
+            // Configure Hadoop job settings. The HadoopFS query is set here.
+            HadoopJob hadoopJob = HadoopJob.newBuilder()
+                    .setMainClass("org.apache.hadoop.fs.FsShell")
+                    .addAllArgs(stringToList(hadoopFSQuery))
+                    .build();
+
+            Job job = Job.newBuilder().setPlacement(jobPlacement).setHadoopJob(hadoopJob).build();
+
+            // Submit an asynchronous request to execute the job.
+            OperationFuture<Job, JobMetadata> submitJobAsOperationAsyncRequest =
+                    jobControllerClient.submitJobAsOperationAsync(projectId, region, job);
+
+            Job response = submitJobAsOperationAsyncRequest.get();
+
+            // Print output from Google Cloud Storage
+            Matcher matches = Pattern.compile("gs://(.*?)/(.*)").matcher(response.getDriverOutputResourceUri());
+            matches.matches();
+
+            Storage storage = StorageOptions.getDefaultInstance().getService();
+            Blob blob = storage.get(matches.group(1), String.format("%s.000000000", matches.group(2)));
+
+            System.out.println(String.format("Job \"%s\" finished: %s",
+                    response.getReference().getJobId(),
+                    new String(blob.getContent())));
+
```

Comment:
if job submission succeed, maybe we could just print out "Job successfully submitted" and check this in the IT.

Reply:
I can remove the job id, but the content of the output is vital for this use case.